File: ChatGPT(예진)/Cpp/달력프로그램.cpp
#include <iostream>
#include <string>

#define MAX_MONTH_NAME_LENGTH 20

void show_month_calendar(int year, int month, int day);
void show_day_calendar(int year, int month, int day);
void choose_day(int year, int month, int days_in_month);
const std::string get_birth_flower(int month);

int main() {
    std::string choice;
    int year, month, day;

    while (true) {
        std::cout << "1. 년도 선택\n";
        std::cout << "2. 달 선택\n";
        std::cout << "3. 종료\n";
        std::cout << "선택하세요: ";
        std::cin >> choice;

        if (choice == "1") {
            std::cout << "년도를 입력하세요: ";
            std::cin >> year;
            month = 0;
            day = 0;
            show_month_calendar(year, month, day);
        }
        else if (choice == "2") {
            std::cout << "년도를 입력하세요: ";
            std::cin >> year;
            std::cout << "달을 입력하세요: ";
            std::cin >> month;
            day = 0;
            show_day_calendar(year, month, day);
        }
        else if (choice == "3") {
            std::cout << "프로그램을 종료합니다.\n";
            break;
        }
        else {
            std::cout << "잘못된 선택입니다. 다시 선택해 주세요.\n";
        }
        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n'); // clear input buffer
    }

    return 0;
}

void show_month_calendar(int year, int month, int day) {
    std::cout << "달을 선택하세요:\n";
    std::string month_names[12] = {
        "January", "February", "March", "April", "May", "June", "July",
        "August", "September", "October", "November", "December"
    };

    for (int i = 0; i < 12; i++) {
        std::cout << i + 1 << ". " << month_names[i] << '\n';
    }

    std::cout << "선택하세요: ";
    std::cin >> month;
    if (month < 1 || month > 12) {
        std::cout << "잘못된 입력입니다.\n";
        return;
    }

    int days_in_month;
    if (month == 2) {
        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            days_in_month = 29;
        }
        else {
            days_in_month = 28;
        }
    }
    else if (month == 4 || month == 6 || month == 9 || month == 11) {
        days_in_month = 30;
    }
    else {
        days_in_month = 31;
    }

    std::cout << "Calendar for " << year << ' ' << month_names[month - 1] << '\n';
    choose_day(year, month, days_in_month);
}

void show_day_calendar(int year, int month, int day) {
    std::cout << "날짜를 선택하세요:\n";
    int days_in_month;
    if (month == 2) {
        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            days_in_month = 29;
        }
        else {
            days_in_month = 28;
        }
    }
    else if (month == 4 || month == 6 || month == 9 || month == 11) {
        days_in_month = 30;
    }
    else {
        days_in_month = 31;
    }

    for (int i = 1; i <= days_in_month; i++) {
        std::cout << i << '\n';
    }

    std::cout << "선택하세요: ";
    std::cin >> day;
    if (day < 1 || day > days_in_month) {
        std::cout << "잘못된 입력입니다.\n";
        return;
    }

    std::cout << year << "년 " << month << "월 " << day << "일\n";
    std::cout << "탄생화: " << get_birth_flower(month) << '\n';
}

void choose_day(int year, int month, int days_in_month) {
    int day;
    std::cout << "날짜를 선택하세요: ";
    std::cin >> day;
    if (day < 1 || day > days_in_month) {
        std::cout << "잘못된 입력입니다.\n";
        return;
    }

    std::cout << year << "년 " << month << "월 " << day << "일\n";
    std::cout << "탄생화: " << get_birth_flower(month) << '\n';
}

const std::string get_birth_flower(int month) {
    const std::string birth_flowers[] = { "", "양귀비", "수선화", "민들레", "벚꽃", "장미", "라벤더", "수련", "해바라기", "백합", "금잔화", "카네이션", "포인세티아" };
    return birth_flowers[month];
}

File: ChatGPT(예진)/Java/달력프로그램.java
import java.util.Scanner;

public class 달력프로그램 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("1. 년도 선택");
            System.out.println("2. 달 선택");
            System.out.println("3. 종료");
            System.out.print("선택하세요: ");
            String choice = scanner.nextLine();

            if (choice.equals("1")) {
                System.out.print("년도를 입력하세요: ");
                int year = scanner.nextInt();
                int month = 0;
                int day = 0;
                showMonthCalendar(year, month, day);
            } else if (choice.equals("2")) {
                System.out.print("년도를 입력하세요: ");
                int year = scanner.nextInt();
                System.out.print("달을 입력하세요: ");
                int month = scanner.nextInt();
                int day = 0;
                showDayCalendar(year, month, day);
            } else if (choice.equals("3")) {
                System.out.println("프로그램을 종료합니다.");
                break;
            } else {
                System.out.println("잘못된 선택입니다. 다시 선택해 주세요.");
            }
            scanner.nextLine(); // Consume newline left-over
        }
        scanner.close();
    }

    public static void showMonthCalendar(int year, int month, int day) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("달을 선택하세요:");
        for (int i = 1; i <= 12; i++) {
            System.out.println(i + ". " + getMonthName(i));
        }

        System.out.print("선택하세요: ");
        month = scanner.nextInt();
        if (month < 1 || month > 12) {
            System.out.println("잘못된 입력입니다.");
            return;
        }

        int daysInMonth = getDaysInMonth(year, month);
        System.out.println(getCalendar(year, month));
        chooseDay(year, month, daysInMonth);
        scanner.close();
    }

    public static void showDayCalendar(int year, int month, int day) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("날짜를 선택하세요:");
        int daysInMonth = getDaysInMonth(year, month);
        for (int i = 1; i <= daysInMonth; i++) {
            System.out.println(i);
        }

        System.out.print("선택하세요: ");
        day = scanner.nextInt();
        if (day < 1 || day > daysInMonth) {
            System.out.println("잘못된 입력입니다.");
            return;
        }

        System.out.println(year + "년 " + month + "월 " + day + "일");
        System.out.println("탄생화: " + getBirthFlower(month));
        scanner.close();
    }

    public static void chooseDay(int year, int month, int daysInMonth) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("날짜를 선택하세요: ");
        int day = scanner.nextInt();
        if (day < 1 || day > daysInMonth) {
            System.out.println("잘못된 입력입니다.");
            return;
        }

        System.out.println(year + "년 " + month + "월 " + day + "일");
        System.out.println("탄생화: " + getBirthFlower(month));
        scanner.close();
    }

    public static String getBirthFlower(int month) {
        String[] birthFlowers = {
                "양귀비", "수선화", "민들레", "벚꽃",
                "장미", "라벤더", "수련", "해바라기",
                "백합", "금잔화", "카네이션", "포인세티아"
        };
        if (month >= 1 && month <= 12) {
            return birthFlowers[month - 1];
        } else {
            return "알 수 없음";
        }
    }

    public static int getDaysInMonth(int year, int month) {
        java.util.Calendar calendar = java.util.Calendar.getInstance();
        calendar.set(java.util.Calendar.YEAR, year);
        calendar.set(java.util.Calendar.MONTH, month - 1);
        return calendar.getActualMaximum(java.util.Calendar.DAY_OF_MONTH);
    }

    public static String getCalendar(int year, int month) {
        java.util.Calendar cal = java.util.Calendar.getInstance();
        cal.set(java.util.Calendar.YEAR, year);
        cal.set(java.util.Calendar.MONTH, month - 1);
        cal.set(java.util.Calendar.DAY_OF_MONTH, 1);

        StringBuilder sb = new StringBuilder();
        sb.append(cal.getDisplayName(java.util.Calendar.MONTH, java.util.Calendar.LONG, java.util.Locale.getDefault()));
        sb.append(" ");
        sb.append(year);
        sb.append("\n");
        sb.append("일  월  화  수  목  금  토\n");

        int dayOfWeek = cal.get(java.util.Calendar.DAY_OF_WEEK);
        for (int i = 1; i < dayOfWeek; i++) {
            sb.append("    ");
        }

        int maxDay = cal.getActualMaximum(java.util.Calendar.DAY_OF_MONTH);
        for (int i = 1; i <= maxDay; i++) {
            sb.append(String.format("%2d", i));
            sb.append("  ");
            if (dayOfWeek % 7 == 0) {
                sb.append("\n");
            }
            dayOfWeek++;
        }
        sb.append("\n");
        return sb.toString();
    }

    public static String getMonthName(int month) {
        java.text.DateFormatSymbols dfs = new java.text.DateFormatSymbols();
        return dfs.getMonths()[month - 1];
    }
}
